### ISSUE TITLE
CRAM reference search order doesn't include 'embed_ref`

### ISSUE THREAD

**Issue by tseemann** [samtools/samtools#720]
http://www.htslib.org/doc/samtools.html says:
```
The search order to obtain a reference is:

Use any local file specified by the command line options (eg -T).
Look for MD5 via REF_CACHE environment variable.
Look for MD5 in each element of the REF_PATH environment variable.
Look for a local file listed in the UR: header tag.
```
Is it safe to assume that if my CRAM was created with `embed_ref=1` then it will use that first?

**Comment by jkbonfield** [comment]
Yes that is a good point.  It should only search for the reference if it's not already embedded within the file itself.

There are some questions I  have to investigate regarding embed_ref though before documenting it.  The format only permits one reference per slice, but in some situations it is beneficial to encode data for multiple references in the same slice, to avoid creating too many slices (name sorted data is the obvious one, but this also crops up with highly fragmented assemblies).  I cannot recall how I dealt with those corner cases.

**Comment by tseemann** [comment]
Is this the only documentation on "slices" ?  I don't understand what they do/are.

```
multi_seq_per_slice=0|1
CRAM output only; defaults to 0 (off). By default CRAM generates one container per reference sequence, except in the case of many small references (such as a fragmented assembly).

seqs_per_slice=INT
CRAM output only; defaults to 10000.

slices_per_container=INT
CRAM output only; defaults to 1. The effect of having multiple slices per container is to share the compression header block between multiple slices. This is unlikely to have any significant impact unless the number of sequences per slice is reduced. (Together these two options control the granularity of random access.)
```

**Comment by jkbonfield** [comment]
See pic5 at the top of page 6 of the CRAM spec: https://github.com/samtools/hts-specs/blob/master/CRAMv3.pdf

I tihnk the defaults are probably fine for you unless you intend on doing a lot of fine grained random access, in which case seqs_per_slice could be reduced (eg to 1000).  I think I concluded it's 5-10% larger with 10fold reduction in slice size.  (Conversely larger slice sizes don't get much beyond 2% smaller.)

However feel free to experiment!  I know you'll report any bugs :)
